### DISCUSSION TITLE
Feature request: Support container+conda ad-hoc combination for environment management

### DISCUSSION THREAD

**Discussion by ManavalanG** [nextflow-io/nextflow#3708] [Q&A]
Hi, this is a feature request based on [slack discussion](https://nfcore.slack.com/archives/CE6SDBX2A/p1677264444061199) with @ewels.

Snakemake can [natively build conda env within a singularity container](https://snakemake.readthedocs.io/en/stable/snakefiles/deployment.html#ad-hoc-combination-of-conda-package-management-with-containers). That is, provided a conda environment and singularity container, snakemake would create conda environment using the singularity container and then use singularity+conda env container for the jobs/processes. I would like to request for similar feature in nextflow. While nextflow supports conda or container usage, it doesn't support container+conda setup.

**Comment by pditommaso** [comment]
Nextflow does the same (and much more) by using [Wave containers](https://www.nextflow.io/docs/latest/wave.html#build-conda-based-containers) service

**Comment by ManavalanG** [comment]
I should have added snakemake does everything locally without having to depend on external services. 

I re-read wave docs closely and it does appear to support this feature. I will report back when I get an opportunity to try it out.
